compiler/src/dotty/tools/backend/sjs/JSEncoding.scala

```diff
@@ -92,6 +92,17 @@ object JSEncoding {
         returnLabelName = Some(freshName("_return"))
       js.Ident(returnLabelName.get)
     }
+
+    /** If the evaluation of `body` adds a `returnLabelName`, wrap the resulting js.Tree to use that label */
+    def makeLabeledIfRequiresEnclosingReturn(tpe: jstpe.Type)(body: js.Tree)(implicit pos: ir.Position): js.Tree = {
+      val inner = body
```

Comment:
This `val` is not needed anymore, now that `body` is by-value. You can directly use `body` inside the cases of the match.